crawl-ref/source/mon-spell.h

```diff
@@ -1133,6 +1133,12 @@ static const mon_spellbook mspell_list[] =
       }
     },
 
+    { MST_STARFLOWER,
+      {
+       { SPELL_HARPOON_SHOT, 80, MON_SPELL_NATURAL },
```

Comment:
You said they were harpooning too often. The number 80 is an out of 200 chance they decide to cast their "spell" on a given turn. Try a lower number, like 20 or 33, so they get down to business if they can melee the player. 

If that's not enough we could add some mon_spell_logic to harpoon shot to not use it if we can melee the target.

Reply:
i'm into the logic idea! i'll be on a train soon